be/src/runtime/http_result_writer.cpp

```diff
@@ -52,7 +52,8 @@ Status HttpResultWriter::_transform_row_to_json(const Columns& result_columns, i
         std::string row;
         ASSIGN_OR_RETURN(row, cast_type_to_json_str(result_column, idx));
         _row_str.append(row);
-        if (result_column != result_columns[num_columns - 1]) {
+        if (&result_column != &(result_columns[num_columns - 1])) {
```

Comment:
Will there be any problem if different columns are later supported to reference the same column through SharedPtr?

Reply:
done